benchmarks/domains/synthetic_2C1D_1C.py

```diff
@@ -0,0 +1,115 @@
+"""Synthetic function with two continuous and one discrete input."""
+
+from uuid import UUID
+
+import numpy as np
+from numpy import pi, sin, sqrt
+from pandas import DataFrame
+
+from baybe.campaign import Campaign
+from baybe.parameters import NumericalContinuousParameter, NumericalDiscreteParameter
+from baybe.recommenders.pure.nonpredictive.sampling import RandomRecommender
+from baybe.searchspace import SearchSpace
+from baybe.simulation import simulate_scenarios
+from baybe.targets import NumericalTarget, TargetMode
+from benchmarks.definition import Benchmark
+from benchmarks.definition.config import (
+    DEFAULT_RECOMMENDER,
+    RecommenderConvergenceAnalysis,
+)
+
+
+def lookup(z: np.ndarray, x: np.ndarray, y: np.ndarray) -> np.ndarray:
+    """Lookup function.
+
+    Inputs:
+        z   discrete   {1,2,3,4}
+        x   continuous [-2*pi, 2*pi]
+        y   continuous [-2*pi, 2*pi]
+    Output: continuous
+    Best value: 4.09685
+    """
+    try:
+        assert np.all(-2 * pi <= x) and np.all(x <= 2 * pi)
+        assert np.all(-2 * pi <= y) and np.all(y <= 2 * pi)
+        assert np.all(np.isin(z, [1, 2, 3, 4]))
+    except AssertionError:
+        raise ValueError("Inputs are not in the valid ranges.")
+
+    return (
+        (z == 1) * sin(x) * (1 + sin(y))
+        + (z == 2) * (x * sin(0.9 * x) + sin(x) * sin(y))
+        + (z == 3) * (sqrt(x + 8) * sin(x) + sin(x) * sin(y))
+        + (z == 4) * (x * sin(1.666 * sqrt(x + 8)) + sin(x) * sin(y))
+    )
+
+
+def benchmark_callable(scenario_config: RecommenderConvergenceAnalysis) -> DataFrame:
+    """Run different recommenders against the test function."""
+    parameters = [
+        NumericalContinuousParameter("x", (-2 * pi, 2 * pi)),
+        NumericalContinuousParameter("y", (-2 * pi, 2 * pi)),
+        NumericalDiscreteParameter("z", (1, 2, 3, 4)),
+    ]
+
+    objective = NumericalTarget(name="target", mode=TargetMode.MAX).to_objective()
+
+    scenarios: dict[str, Campaign] = {}
+    for scenario_name, recommender in scenario_config.recommenders.items():
+        recommender_dct = (
+            {} if recommender is DEFAULT_RECOMMENDER else {"recommender": recommender}
+        )
+        campaign = Campaign(
+            searchspace=SearchSpace.from_product(parameters=parameters),
+            objective=objective,
+            **recommender_dct,
+        )
+        scenarios[scenario_name] = campaign
+
+    return simulate_scenarios(
+        scenarios,
+        lookup,
+        batch_size=scenario_config.batch_size,
+        n_doe_iterations=scenario_config.n_doe_iterations,
+        n_mc_iterations=scenario_config.n_mc_iterations,
+        impute_mode="error",
+    )
+
+
+benchmark_config = RecommenderConvergenceAnalysis(
+    recommenders={
+        "Default Recommender": DEFAULT_RECOMMENDER,
```

Comment:
theres toerh ebnchmarks where we will not wan to parameterize over recommenders, but over the way parameters are encoded or various other things. If we do this inside the callable then this will be multiplied with the reocmmedners here and also that feels liek a strange different treatment between the way we parameteriz eover different types of settings
how do you envision eg the direct arylation (full lookup) benchmark specified in this benchmark?

Reply:
The current class was just my way of generalizing what @fabianliebig hardcoded for the first example. We can certainly discuss if/how to keep it. For the arylation-type of benchmark, I see two options:
* Either, we keep the current config class as is to make it only describe recommender changes. We could then add a separate class tailored to parameter configuration changes. Note also that even the class approach itself is completely optional, we can pass as configuration whatever object we like.
* Or, we generalize the current setting to the point of the common denominator of the two use cases, which would be the replacing the `recommenders` attribute with a more general `campaigns` attribute.
* Or, any of your suggestions. Anything in mind?

Reply:
My vote is still for "Let's not make all of this super configurable but just define everything in the callable, making this one callable well defined", but I know that I am the only one with this opinion 😉 
If we want to stay with this approach, then I'd propose to split it up. There are things that need to be configurable in every benchmark (`batch_size`, `n_dor_iterations`, `n_mc_iteration`) and aspects that we might want to only vary in individual benchmarks (`recommenders`, `encodings`) and so on. I'd propose to have separate classes for the latter: One `Analysis` class that only changes `recommenders`, one for `encoding` and so on. For me, this feels like the cleanest approach when we want to make things configurable, as having just a `campaign` is just too general in my opinion.

(Still, I wonder how many different benchmarks we want to have in the end, and if crafting a dedicated structure for making stuff configurable is actually necessary....)

Reply:
I also see it like that, I'd be fine if encodings, recommenders etc are all responsibilities of the callable. But in that case this configurability of the recommenders here is inconsistent and needs to be removed